### DISCUSSION TITLE
Use it with laravel media library package

### DISCUSSION THREAD

**Discussion by allaghi** [rahulhaque/laravel-filepond#34] [Q&A] [answered]
Hi, 

I am wondering if is there any a proper way to use it with laravel media library package.

Thanks in advance .

**Comment by rahulhaque** [comment, marked as answer]
> I am wondering if is there any a proper way to use it with laravel media library package.

Yes, there is.

```php
// Single file
$model->addMedia(Filepond::field($request->image)->getFile())->usingName('...')->toMediaCollection('...');

// Multiple files
$images = Filepond::field($request->images)->getFile();
foreach ($images as $image) {
    $model->addMedia($image)->usingName('...')->toMediaCollection('...');
}
```

I recommend using `preserveOriginal()` if you want the temporary files to be available when used with soft delete enabled.

**Reply by allaghi** [reply]
When I use Filepond::field with multiple I got this error, 

RahulHaque\Filepond\AbstractFilepond::decrypt(): Argument #1 ($data) must be of type string, array given, called in C:\xampp\htdocs\laravel9-breeze\vendor\rahulhaque\laravel-filepond\src\AbstractFilepond.php on line 51


Note , I am using inertia with it as your example

**Reply by allaghi** [reply]
$files = Filepond::field(Request::input('gallery'))->getFile();

**Reply by rahulhaque** [reply]
Did you run my example? Is multiple upload working fine there? If not, open an issue there and I'll fix. If you're facing issues with your custom code, please share the code so that I can reproduce this issue and suggest a fix.

**Reply by allaghi** [reply]
I did the same your steps in my own project and when I use $files = Filepond::field(Request::input('gallery'))->getFile();
with multiple upload I got 

RahulHaque\Filepond\AbstractFilepond::decrypt(): Argument https://github.com/rahulhaque/laravel-filepond/issues/1 ($data) must be of type string, array given, called in C:\xampp\htdocs\laravel9-breeze\vendor\rahulhaque\laravel-filepond\src\AbstractFilepond.php on line 51

I don't know why ?

**Reply by allaghi** [reply]
I don't know if I missed something ?

**Reply by allaghi** [reply]
Finally it works , as I said I missed something , Thank you so much for your reply and your package , its amazing.

**Reply by rahulhaque** [reply]
Thanks a lot. Please mark the answer as accepted.
